qualtran/protos/data_types.proto

```diff
@@ -0,0 +1,62 @@
+/*
+  Copyright 2023 Google LLC
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+syntax = "proto3";
+
+package qualtran;
+
+message QDType {
+    oneof bitsize {
+        int64 bitsize_int = 1;
+        string bitsize_expr = 2;
+    }
```

Comment:
As discussed during our call, I think it would make more sense to have the `bitsize` argument in the individual data types; more closely matching the structure of the library. 

```suggestion
```